### PR TITLE
Issue #78 Add preflight code for `/etc/NetworkManager/dnsmasq.d/crc.conf`

### LINKED ISSUE #78: Add a preflight check for `/etc/NetworkManager/dnsmasq.d/crc.conf` for linux
We are right now not checking if `/etc/NetworkManager/dnsmasq.d/crc.conf` exist and network manager using it, this should be part of preflight checks.

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #78

### CODE REVIEW

**Comment by gbraad** [comment]
:+1:  Thanks!

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 1]
```diff
@@ -456,3 +461,34 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
...
+	_, err := os.Stat(crcDnsmasqConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("File not found: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	config, err := ioutil.ReadFile(crcDnsmasqConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("Error opening file: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	if len(config) != len(c) {
+		return false, fmt.Errorf("File seems corrupt: %s: %s", crcDnsmasqConfigPath)
```
Comment (gbraad) [inline pkg/crc/preflight/preflight_checks_linux.go]: interesting conclusion...

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 2]
```diff
@@ -27,10 +27,15 @@ const (
 	libvirtDriverCommand     = "crc-driver-libvirt"
 	libvirtDriverDownloadURL = "https://github.com/code-ready/machine-driver-libvirt/releases/download/0.9.1/crc-driver-libvirt"
 	defaultPoolSize          = "20 GB"
+	crcDnsmasqConfigFile     = "crc.conf"
 )
 
 var (
 	libvirtDriverBinaryPath = filepath.Join(driverBinaryDir, libvirtDriverCommand)
+	crcDnsmasqConfigPath    = filepath.Join("/", "etc", "NetworkManager", "dnsmasq.d", crcDnsmasqConfigFile)
```
Comment (praveenkumar) [inline pkg/crc/preflight/preflight_checks_linux.go]: Don't use hard coded separator. `filepath.Join(string(filepath.Separator),"A","B","C")`

- https://golang.org/pkg/path/filepath/#pkg-constants

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 3]
```diff
@@ -456,3 +461,34 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
...
+		return false, fmt.Errorf("File not found: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	config, err := ioutil.ReadFile(crcDnsmasqConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("Error opening file: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	if len(config) != len(c) {
+		return false, fmt.Errorf("File seems corrupt: %s: %s", crcDnsmasqConfigPath)
+	}
+	if !bytes.Equal(config, c) {
```
Comment (praveenkumar) [inline pkg/crc/preflight/preflight_checks_linux.go]: If we are already comparing the bytes the why we need `len` also?

Comment (gbraad) [reply pkg/crc/preflight/preflight_checks_linux.go]: same thought here...

File: pkg/crc/preflight/preflight_checks_linux.go (hunk 2 above)
Comment (gbraad) [reply pkg/crc/preflight/preflight_checks_linux.go]: well, this is already Linux-only, but sure.

**Comment by gbraad** [comment]
Since @anjannath is on PTO I will make the necessary changes.

**Comment by gbraad** [comment]
Close this in favour of #92

**Pull request closed without merging**